function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/Synapse/OlSynapseMessageConsolidation.cs

```diff
@@ -1,59 +1,124 @@
+                    return _blobClientFactory.UploadAsync(CONTAINER_NAME, outputBlobNames[index], outputJson);
+                }));
+
+                // Await all upload tasks
+                await Task.WhenAll(inputUploadTasks.Concat(outputUploadTasks));
+
+
+                // Get list of input and outputs which are stored in blob
+                List<string> inputs = await _blobClientFactory.GetBlobsByHierarchyAsync(prefixInput);
+                List<string> outputs = await _blobClientFactory.GetBlobsByHierarchyAsync(prefixOutput);
+
+                // If there is at least 1 input and 1 output - proceed for message consolidation
+                if (inputs.Count > 0 && outputs.Count > 0)
+                {
+                    List<Inputs?> inputsList = new List<Inputs?>();
+                    List<Outputs?> outputsList = new List<Outputs?>();
+
+                    // Get all the inputs
+                    List<Task<Inputs?>> inputTasks = inputs.Select(async item =>
+                    {
+                        var inputObject = await _blobClientFactory.DownloadBlobAsync(CONTAINER_NAME, item);
+                        return JsonConvert.DeserializeObject<Inputs?>(inputObject);
+                    }).ToList();
+
+                    // Get all the outputs
+                    List<Task<Outputs?>> outputTasks = outputs.Select(async item =>
+                    {
+                        var outputObject = await _blobClientFactory.DownloadBlobAsync(CONTAINER_NAME, item);
+                        return JsonConvert.DeserializeObject<Outputs?>(outputObject);
+                    }).ToList();
+
+                    inputsList = (await Task.WhenAll(inputTasks)).Where(input => input != null).ToList();
+                    outputsList = (await Task.WhenAll(outputTasks)).Where(output => output != null).ToList();
+
+                    // Message Consolidation - Update inputs / outputs to the current olevent object
+                    if (inputsList.Count > 0)
+                    {
+                        olEvent.Inputs = inputsList;
+                    }
+
+                    if (outputsList.Count > 0)
+                    {
+                        olEvent.Outputs = outputsList;
+                    }
+                    // TO DO check mani - parallel foreach
+                    // TO DO check mani -  any uniqueness in input and output
+                    // if we have same event same i/o it will add , any error further
+                    // Do we need to do distinct on names of input or output etc
+                    return olEvent;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            catch (Exception ex)
+            {
+                _log.LogError(ex, $"OlSynapseMessageConsolidation-ConsolidateEventAsync: Error {ex.Message} ");
+                throw;  // TO DO check mani - throw if there is any error?
```

Comment:
- Allow the logic to raise the exception as the outer handler will fail gracefully
- consider the following approach as a best practice for structured logging: _log.LogError(ex, "...: Error {errorMessage}", ex.Message);

Reply:
Done